src/routes/Chat/container.js

```diff
@@ -65,6 +65,13 @@ export class ChatContainer extends Component {
 		this.loadMoreMessages();
 	}
 
+	handleChangeText = async(text) => {
+		await this.grantUser();
```

Comment:
I suggest you to read the `user` and `room` objects from `this.props`, avoiding unnecessary REST endpoints calls.
If the `user` or the `room` are not assigned, just return.